lib/src/components/NxPolicyViolationIndicator/__tests__/NxPolicyViolationIndicator.test.tsx

```diff
@@ -4,47 +4,89 @@
-    expect(getShallowComponent()).toMatchSelector('.nx-policy-violation-indicator--unspecified');
+  it('adds specified classnames in addition to the defaults', function() {
+    const defaultEl = renderEl()!,
+        el = renderEl({ className: 'foo' })!;
+
+    expect(el).toHaveClass('foo');
+
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('sets the modifier class using the threatLevelCategory if it is provided', function() {
-    expect(getShallowComponent({ threatLevelCategory: 'low' })).toMatchSelector('.nx-policy-violation-indicator--low');
-    expect(getShallowComponent({ threatLevelCategory: 'low' }))
-        .not.toMatchSelector('.nx-policy-violation-indicator--unspecified');
+  it('adds additional attributes to the top-level element', function() {
+    const el = renderEl({ id: 'foo', lang: 'en' });
+
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('lang', 'en');
   });
 
-  it('sets the modifier class by converting the policyThreatLevel if it is provided', function() {
-    expect(getShallowComponent({ policyThreatLevel: 0 })).toMatchSelector('.nx-policy-violation-indicator--none');
-    expect(getShallowComponent({ policyThreatLevel: 1 })).toMatchSelector('.nx-policy-violation-indicator--low');
-    expect(getShallowComponent({ policyThreatLevel: 3 })).toMatchSelector('.nx-policy-violation-indicator--moderate');
-    expect(getShallowComponent({ policyThreatLevel: 5 })).toMatchSelector('.nx-policy-violation-indicator--severe');
-    expect(getShallowComponent({ policyThreatLevel: 9 })).toMatchSelector('.nx-policy-violation-indicator--critical');
-    expect(getShallowComponent({ policyThreatLevel: 9 }))
-        .not.toMatchSelector('.nx-policy-violation-indicator--unspecified');
+  it('applies a ref to the top level element', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        el = renderEl({ref: ref})!;
+
+    expect(ref.current).toBe(el);
+  });
+
+  it('sets the text according to threatLevelCategory if no text is provided', function() {
+    const { container, rerender } = render(
+      <NxPolicyViolationIndicator threatLevelCategory='none'></NxPolicyViolationIndicator>
+    );
+    expect(container).toHaveTextContent('none');
+
+    rerender(<NxPolicyViolationIndicator threatLevelCategory='moderate'></NxPolicyViolationIndicator>);
+    expect(container).toHaveTextContent('moderate');
   });
 
-  it('sets the modifier class using the threatLevelCategory if both props are provided', function() {
-    const component = getShallowComponent({ policyThreatLevel: 9, threatLevelCategory: 'low' });
+  it('overrides the default threatLevelCategory text if text is provided', function() {
+    const { container } = render(
+      <NxPolicyViolationIndicator threatLevelCategory='low'>Foo</NxPolicyViolationIndicator>
+    );
```

Comment:
```suggestion
    const el = renderEl({ threatLevelCategory: 'low', children: 'Foo' })!;
```
And then use `el` in the expectations not `container`

Reply:
>And then use el in the expectations not container

fixed here: [6adb242](https://github.com/sonatype/sonatype-react-shared-components/pull/911/commits/6adb242c1c0584669ac258fa1bc92d11b9dba698)